tasks/bdx.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  getAllBDStables,
+  getBdUs,
+  getBdx,
+  getOperationalTreasury,
+  getStakingRewardsDistribution,
+  getTreasury
+} from "../utils/DeployedContractsHelpers";
+import { bigNumberToDecimal } from "../utils/NumbersHelpers";
+import { getAllUniswapPairsData } from "./liquidity-pools";
+
+export function load() {
+  task("bdx:show:diagnostics").setAction(async (args, hre) => {
+    const bdx = await getBdx(hre);
+    let bdxTotalSupply = await bdx.totalSupply();
+    console.log("BDX Total Supply:", bigNumberToDecimal(bdxTotalSupply, 18).toLocaleString());
+
+    const [treasury, srd, stables] = await Promise.all([getTreasury(hre), getStakingRewardsDistribution(hre), getAllBDStables(hre)]);
+    const operationalTreasuryAddress = hre.network.name === "rsk" ? rskOperationalTreasuryAddress : (await getOperationalTreasury(hre)).address;
+
+    for (const stable of stables) {
+      console.log(`${await stable.symbol()}: -${bigNumberToDecimal(await bdx.balanceOf(stable.address), 18).toLocaleString()}`);
+      bdxTotalSupply = bdxTotalSupply.sub(await bdx.balanceOf(stable.address));
+    }
+
+    const [treasuryBalance, srdBalance, operationalTreasuryBalance] = await Promise.all([
+      bdx.balanceOf(treasury.address),
+      bdx.balanceOf(srd.address),
+      bdx.balanceOf(operationalTreasuryAddress)
+    ]);
+
+    console.log(`Treasury: -${bigNumberToDecimal(treasuryBalance, 18).toLocaleString()}`);
+    console.log(`Operational Treasury: -${bigNumberToDecimal(operationalTreasuryBalance, 18).toLocaleString()}`);
+    console.log(`Staking Rewards Distribution (anything in staking/vesting): -${bigNumberToDecimal(srdBalance, 18).toLocaleString()}`);
+
+    bdxTotalSupply = bdxTotalSupply.sub(treasuryBalance).sub(srdBalance).sub(operationalTreasuryBalance);
+
+    if (hre.network.name === "rsk") {
+      const lockingContractBalance = await bdx.balanceOf(bdxLockingContractAddressRSK);
+      bdxTotalSupply = bdxTotalSupply.sub(lockingContractBalance);
+      console.log(`Locking contract: -${bigNumberToDecimal(lockingContractBalance, 18).toLocaleString()}`);
+    }
+
+    async function getBdxInUniswapPools(hre: HardhatRuntimeEnvironment, bdx: BDXShares) {
+      const uniswapPairsData = await getAllUniswapPairsData(hre);
+      const bdxSymbol = await bdx.symbol();
+      let bdxInPools = 0;
+
+      for (const bdxPair of uniswapPairsData.filter(pairData => pairData.isBdxPool)) {
+        const bdxReserves = (
+          bdxPair[bdxSymbol] as {
+            address: string;
+            decimals: number;
+            reserves: number;
+          }
+        ).reserves;
+        bdxInPools += bdxReserves;
+
+        console.log(`- ${bdxPair.pairName} uniswap pool: ${bdxReserves.toLocaleString()}`);
+      }
+
+      console.log(`\n- Total in the pools: ${bdxInPools.toLocaleString()}`);
+      return bdxInPools;
+    }
+
+    const bdxTotalSupply_d18 = bigNumberToDecimal(bdxTotalSupply, 18);
```

Comment:
your naming is the other way around:
* `bdxTotalSupply_d18` should be `bdxTotalSupply` (it's a normal number)
* `bdxTotalSupply` should be bdxTotalSupply_d18 (it's a BidNumber with 18 digits of precision)

also insead of `bigNumberToDecimal(x, 18)` you can use `d18_ToNumber(bdxTotalSupply);`

Reply:
my bad, nice catch!